CONTRIBUTING.md

```diff
@@ -1,77 +1,175 @@
+- [website (`gh-pages` branch of this repo)](https://github.com/microsoft/microsoft-ui-xaml/tree/gh-pages)
+- [Xaml Controls Gallery (sample app)](https://github.com/microsoft/Xaml-Controls-Gallery/tree/master)
+
+#### Team Process
+Team process includes feedback about our repo and overall communication. Suggestions for improvement and proposed alternatives are very helpful here. 
+- Ex: "`<`specific aspect of team process - e.g., response frequency, format`>` of the repo is making it hard for me to achieve `<`goal`>`. Please consider doing `<`alternative`>` instead because that would help `<`action`>`" 
+
+## Code Contribution Guidelines
+Note this repository is not ready for open source collaboration at this time; this work is currently in progress. You can track the WinUI team’s progress towards open source collaboration in the [Phased Rollout to Open Source Collaboration](https://github.com/microsoft/microsoft-ui-xaml/discussions/10700) discussion.
+
+WinUI will be taking a phased approach to opening up the repo:
+**Phase 1: Increased Mirror Frequency**
+After the WASDK 1.8 release (end of August), we’ll begin more frequent mirroring of internal commits to GitHub to increase transparency and show progress.
+**Phase 2: 3rd Party Devs Build Locally**
+External developers will be able to clone and build the repo locally, with documentation to guide setup and dependencies.
+**Phase 3: 3rd Party Devs Contribute & Run Tests**
+Contributors will be able to submit PRs and run tests locally. We’re working to untangle private dependencies and make test infrastructure publicly accessible.
+**Phase 4: GitHub as Center of Gravity**
+GitHub becomes the primary place for development, issue tracking, and community engagement. Internal mirrors will be phased out.
+
+### [WIP] New Contributors
+Contributions from the community are greatly appreciated. We mark the most straightforward issues with labels. These issues are the best place to start if you are interested in contributing but are new to the codebase.
+
+- [good first issues](https://github.com/Microsoft/microsoft-ui-xaml/labels/good first issue)
+- [help wanted](https://github.com/orgs/microsoft/projects/1868/views/12)
 
+Another great way to help is by up-voting and commenting on feature proposals:
+- [feature request](https://github.com/Microsoft/microsoft-ui-xaml/labels/feature request)
+
+### [WIP] Code Contribution Process
+We use and recommend the following workflow:
+1. Create an issue for your work. 
+    - You can skip this step for trivial changes or if there is an existing issue for the bug/feature.
+    - If your change adds or changes public APIs or UI, first follow the [New Feature or API Process](docs/feature_proposal_process.md).
+    - Clearly state that you are going to take on implementing it, if that's the case. You can request that the issue be assigned to you. Note: The issue filer and the implementer don't have to be the same person.
```

Comment:
I know this part is WIP, but I don't this guidance is up to date.
```suggestion
1. Have an issue tracking what you want to fix. 
    - If one doesn't already exist, open an issue. But before investing time on a contribution make sure the issue has been triaged and your fix has a chance of being merged.
    - If your change adds or changes public APIs or UI, first follow the [New Feature or API Process](docs/feature_proposal_process.md).
    - Comment on the issue to indicate your intent to work on it.
```

Reply:
Yeah, there's a lot of old guidance scattered everywhere, so there's a few different steps in flight to try and get this up-to-date. Part of it too is whether we document how things work now, or more how we want things to work - and then move towards aligning to that as the OSS plan moves forward.